pennylane/qnode.py

```diff
@@ -616,6 +617,36 @@ def __call__(self, *args, **kwargs):
         )
         self._tape_cached = using_custom_cache and self.tape.hash in cache
 
+        if qml.active_return():
+            res = qml.execute_new(
+                [self.tape],
+                device=self.device,
+                gradient_fn=self.gradient_fn,
+                interface=self.interface,
+                gradient_kwargs=self.gradient_kwargs,
+                override_shots=override_shots,
+                **self.execute_kwargs,
+            )
+
+            res = res[0]
+
+            # Autograd or tensorflow: they do not support tuple return with backpropagation
+            backprop = False
+            if not isinstance(self._qfunc_output, qml.measurements.MeasurementProcess):
+                backprop = any(qml.math.in_backprop(x) for x in res)
+            if self.interface in ("tf", "autograd") and self.gradient_fn == "backprop" and backprop:
+                res = self.device._asarray(res)
+
+            # If the return type is not tuple (list or ndarray)
+            if not isinstance(self._qfunc_output, (tuple, qml.measurements.MeasurementProcess)):
+                if not self.device._shot_vector:
+                    res = type(self.tape._qfunc_output)(res)
+                else:
+                    res = [type(self.tape._qfunc_output)(r) for r in res]
+                    res = tuple(res)

```

Comment:
It seems like this would conflict with the previous few lines concerning backpropagation, for example in the case where `_qfunc_output` is a list, the shot vector is `None`, the interface is tensorflow, and the output requires backprop.

Reply:
You are right it might be problematic for `tf` with `backprop`, because returning a list might prevents us to take the jacobian. I will add a condition here and a test. Thank you!